src/datasets/io/abc.py

```diff
@@ -0,0 +1,26 @@
+from abc import ABC, abstractmethod
+from typing import Optional
+
+from .. import Features, NamedSplit
+from ..arrow_dataset import Dataset
+from ..utils.typing import PathLike
+
+
+class AbstractDatasetBuilder(ABC):
```

Comment:
We've already a DatasetBuilder abstraction.
Maybe we can find another name ?

Or maybe we can simply add `build` to the existing DatasetBuilder

Reply:
Yes, @lhoestq, I totally agree with you. I was thinking of this since I pushed the change: it is confusing to call it Builder, because we have already a builder classes. I am going to rename it back to `DatasetReader.read`.

I prefer having a specific class to deal with `from_csv` method: it is basically an adapter with a single method, which hides all the complexity underneath. I would prefer not adding another method to DatasetBuilder: my long-term objective is in the opposite direction, by reducing that class (which is already too large) by composing it of specialized classes...

Thanks again for your critical view!! :heart:

Reply:
Makes sense !
I agree that DatasetBuilder is already quite big, let's have a DatasetReader then :)

Reply:
Done @lhoestq